packages/react-data-grid/src/HeaderRow.tsx

```diff
@@ -130,9 +136,10 @@ export default class HeaderRow extends React.Component<Props> {
   }
 
   setScrollLeft(scrollLeft: number): void {
-    this.props.columns.forEach((column?: Column, i?: number) => {
-      const cell = this.cells[i!];
-      if (!cell) return;
+    this.props.columns.forEach((column?: Column) => {
+      const key = this.getHeaderCellKey(column!);
+      if (!this.cells.has(key)) return;
```

Comment:
In Javascript I would prefer something like this

```
const cell = this.cells.get(key);
if (!cell) return;
```

Theoretically, finding the key procedure in the the map will just happen once in my code snippet.

Reply:
Sure, but if you're familiar with js engine optimization you'd see that:
- `.has()` returns a boolean, which is faster in an if condition, as the engine doesn't have to cast the object or undefined value to a boolean.
- `cell` is thus monomorphic, so it's easier for the engine to optimize.

https://ponyfoo.com/articles/javascript-performance-pitfalls-v8